src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -627,6 +628,30 @@ private void ReportAssignmentOperatorError(AssignmentExpressionSyntax node, Diag
 
         private static void ReportBinaryOperatorError(ExpressionSyntax node, DiagnosticBag diagnostics, SyntaxToken operatorToken, BoundExpression left, BoundExpression right, LookupResultKind resultKind)
         {
+            if (resultKind == LookupResultKind.Empty && operatorToken.Kind() == SyntaxKind.DotDotToken)
+            {
+                var leftSpecialType = left.Type.SpecialType;
+                var rightSpecialType = right.Type.SpecialType;
+                if (leftSpecialType.IsIntegralType() && rightSpecialType.IsIntegralType())
+                {
+                    // User meant to use the built-in range type, but it was missing
+                    if (leftSpecialType == SpecialType.System_Int64 || leftSpecialType == SpecialType.System_UInt64 ||
+                        rightSpecialType == SpecialType.System_Int64 || rightSpecialType == SpecialType.System_UInt64)
```

Comment:
>SpecialType.System_UInt64 [](start = 92, length = 25)

I don't think the current design is intended to support ranges over ulong. Or uint, for that matter, except by their conversion to long.

Reply:
I think you're correct. The errors should be:

* `ulong`: Same error that occurs when using `float` or `MyType` (`ERR_BadBinaryOps`)
* `long`: Error, missing `LongRange`
* `uint`: Error, missing `LongRange`
* `int`: Error, missing `Range`
* everything below int: Error, missing `Range`